src/main/java/at/hannibal2/skyhanni/features/combat/end/DragonFeatures.kt

```diff
@@ -0,0 +1,377 @@
+     * REGEX-TEST:  §r§bItsJxxxxx2001: §r§c457k❤
+     * REGEX-TEST:  §r§bThunderblade73: §r§c12.3k❤
+     */
+    private val tabDamage by tabListGroup.pattern(
+        "fight.player",
+        ".*§r§.(?<Name>.+): §r§c(?<Damage>[\\d.]+)(?<Unit>[kM])?❤",
+    )
+
+    private var yourEyes = 0
+
+    private var dragonSpawned = false
+        set(value) {
+            field = value
+            if (dragonSpawned) {
+                egg = false
+            }
+        }
+
+    private enum class Type {
+        GOLEM,
+        DRAGON
+    }
+
+    private var endType: Type? = null
+    private var endTopDamage = 0.0
+    private var endDamage = 0.0
+    private var endPlace = 0
+
+    private var currentDamage = 0.0
+    private var currentTopDamage = 0.0
+    private var currentPlace: Int? = null
+    private var widgetActive = false
+    var egg = true
+    var weight = 0.0
+    private var currentDragonType: DragonType? = null
+
+    private fun resetEnd() {
+        endType = null
+        endTopDamage = 0.0
+        endDamage = 0.0
+        endPlace = 0
+    }
+
+    private fun reset() {
+        resetEnd()
+        dragonSpawned = false
+        currentTopDamage = 0.0
+        currentDamage = 0.0
+        currentPlace = null
+        widgetActive = false
+        yourEyes = 0
+        currentDragonType = null
+    }
+
+    private fun enable() = LorenzUtils.inSkyBlock && IslandType.THE_END.isInIsland()
+
+    private fun enableDisplay() = enable() && config.display
+
+    private fun weightMap(place: Int) = when (place) {
+        -1 -> 10
+        1 -> 200
+        2 -> 175
+        3 -> 150
+        4 -> 125
+        5 -> 110
+        6, 7, 8 -> 100
+        9, 10 -> 90
+        11, 12 -> 80
+        else -> 70
+    }
+
+    private fun calculateDragonWeight(eyes: Int, place: Int, firstDamage: Double, yourDamage: Double) =
+        weightMap(
+            if (yourDamage == 0.0) -1 else place,
+        ) + 100 * (
+            eyes + yourDamage / (firstDamage.takeIf { it != 0.0 } ?: 1.0)
+            )
+
+    private fun calculateProtectorWeight(zealots: Int, place: Int, firstDamage: Double, yourDamage: Double) =
+        weightMap(
+            if (yourDamage == 0.0) -1 else place,
+        ) + 50 * (
+            yourDamage / (firstDamage.takeIf { it != 0.0 } ?: 1.0)
+            ) + if (zealots > 100) 100 else zealots
+
+    @Suppress("CyclomaticComplexMethod")
+    @HandleEvent
+    fun onChat(event: SkyHanniChatEvent) {
+        if (!enable()) return
+        val message = event.message
+        if (!config.chat && !config.display && !config.superiorNotify && !configProtector) return
+        dragonSpawn.matchMatcher(message) {
+            dragonSpawned = true
+
+            currentDragonType = DragonType.valueOf(this.group("Dragon").uppercase())
+            ChatUtils.debug("Dragon Type: $currentDragonType")
+
+            if (config.superiorNotify && this.group("Dragon") == "Superior") {
```

Comment:
```suggestion
            if (config.superiorNotify && currentDragonType == DragonType.SUPERIOR) {
```

Please don't match by strings when you have an enum available to do so

Reply:
Ty for noticing :D